### PR TITLE
feat: provide selectable columns as search settings

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-1192

# TODO

- [x] Unit tests
- [x] Validate integration with FE

### CODE REVIEW

File: models/classes/search/SearchProxy.php [hunk 1]
```diff
@@ -100,8 +104,16 @@ public function search(ServerRequestInterface $request): array
...
         $results = $this->executeSearch($query);
 
-        return $this->getResultSetResponseNormalizer()
+        $response = $this->getResultSetResponseNormalizer()
             ->normalize($query, $results, $queryParams['params']['structure']);
+
+        $response['settings'] = $this->getSearchSettingsService()
+            ->getSettingsByClassMetadataSearchRequest(
+                (new ClassMetadataSearchRequest())
+                    ->setClassUri($request->getQueryParams()['params']['parentNode'])
```
Comment (hectoras) [inline models/classes/search/SearchProxy.php:113]: nitpick: You already have a local copy of the query params here, the method call can be avoided

```suggestion
                    ->setClassUri($queryParams['params']['parentNode'])
```

File: models/classes/search/Contract/SearchSettingsServiceInterface.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+namespace oat\tao\model\search\Contract;
+
+use oat\tao\model\Lists\Business\Domain\ClassMetadataSearchRequest;
+use oat\tao\model\search\SearchSettings;
+
+interface SearchSettingsServiceInterface
+{
+    public function getSettingsByClassMetadataSearchRequest(ClassMetadataSearchRequest $classMetadataSearchRequest): SearchSettings;
```
Comment (shpran) [inline models/classes/search/Contract/SearchSettingsServiceInterface.php:30]:
```suggestion
    public function getSettingsByClassMetadataSearchRequest(
        ClassMetadataSearchRequest $classMetadataSearchRequest
    ): SearchSettings;
```

File: models/classes/search/Service/DefaultSearchSettingsService.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace oat\tao\model\search\Service;
+
+use oat\tao\model\Lists\Business\Domain\ClassMetadataSearchRequest;
+use oat\tao\model\search\Contract\SearchSettingsServiceInterface;
+use oat\tao\model\search\SearchSettings;
+
+class DefaultSearchSettingsService implements SearchSettingsServiceInterface
+{
+    public function getSettingsByClassMetadataSearchRequest(ClassMetadataSearchRequest $classMetadataSearchRequest): SearchSettings
```
Comment (shpran) [inline models/classes/search/Service/DefaultSearchSettingsService.php:31]:
```suggestion
    public function getSettingsByClassMetadataSearchRequest(
        ClassMetadataSearchRequest $classMetadataSearchRequest
    ): SearchSettings
```

**Review by shpran — APPROVED** [review]
Good job 👍🏻 
Left some CS comments

- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

**Pull request merged**